index.js

```diff
@@ -48,6 +54,141 @@ class ExecutorQueue extends Executor {
         this.fuseBox = new FuseBox();
         this.fuseBox.addFuse(this.queueBreaker);
         this.fuseBox.addFuse(this.redisBreaker);
+
+        const RETRY_LIMIT = 3;
+        const RETRY_DELAY = 5;
+        const retryOptions = {
+            plugins: ['retry'],
+            pluginOptions: {
+                retry: {
+                    retryLimit: RETRY_LIMIT,
+                    retryDelay: RETRY_DELAY
+                }
+            }
+        };
+        // Jobs object to register the worker with
+        const jobs = {
+            startDelayed: Object.assign({ perform: (jobConfig, callback) =>
+                this.redisBreaker.runCommand('hget', this.periodicBuildTable,
+                    jobConfig.jobId)
+                    .then(fullConfig => this.startPeriodic(JSON.parse(fullConfig), true))
+                    .then(result => callback(null, result), (err) => {
+                        winston.error('err in startDelayed job: ', err);
+                        callback(err);
+                    })
+            }, retryOptions)
+        };
+
+        // eslint-disable-next-line new-cap
+        this.multiWorker = new Resque.multiWorker({
```

Comment:
I think we need to have at least some logging for failure or error, otherwise it will be very hard to debug: https://github.com/screwdriver-cd/queue-worker/blob/master/index.js

Reply:
Added all the logging events for now, will remove based on need